qiskit/result/counts.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A container class for counts from a circuit execution."""
+
+import re
+
+from qiskit.result import postprocess
+from qiskit import exceptions
+
+
+# NOTE: A dict subclass should not overload any dunder methods like __getitem__
+# this can cause unexpected behavior and issues as the cPython dict
+# implementation has many standard methods in C for performance and the dunder
+# methods are not always used as expected. For example, update() doesn't call
+# __setitem__ so overloading __setitem__ would not always provide the expected
+# result
+class Counts(dict):
+    """A class to store a counts result from a circuit execution."""
+
+    def __init__(self, data, name=None, shots=None, time_taken=None,
+                 creg_sizes=None, memory_slots=None):
+        """Build a counts object
+
+        Args:
+            data (dict): The dictionary input for the counts. Where the keys
+                represent a measured classical value and the value is an
+                integer the number of shots with that result.
+                The keys can be one of several formats:
+
+                     * A hexademical string of the form ``"0x4a"``
+                     * A bit string prefixed with ``0b`` for example ``'0b1011'``
+                     * A bit string formatted across register and memory slots.
+                       For example, ``'00 10'``.
+                     * A dit string, for example ``'02'``. Note for objects created
+                       with dit strings the ``creg_sizes``and ``memory_slots``
+                       kwargs don't work and :meth:`hex_outcomes` and
+                       :meth:`int_outcomes` also do not work.
+
+            name (str): A string name for the counts object
+            shots (int): The number of shots used in the experiment
+            time_taken (float): The duration of the experiment that generated
+                the counts
+            creg_sizes (list): a nested list where the inner element is a list
+                of tuples containing both the classical register name and
+                classical register size. For example,
+                ``[('c_reg', 2), ('my_creg', 4)]``.
+            memory_slots (int): The number of total ``memory_slots`` in the
+                experiment.
+            metadata: Any arbitrary key value metadata passed in as kwargs.
```

Comment:
I think this was removed from init function

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/4501/commits/91cf8fbe192498782fc49656eb0fd2813dc635d8